TESTING.md

```diff
@@ -37,14 +37,42 @@ Documentation located [here](./e2e/README.md)
 
 ## Code Coverage
 
-* 100% statement coverage is achievable and desirable.
+It's up to the individual developer as to whether they want to add line coverage in the form of a unit test or e2e test.
 
-Codecov.io will combine each of the above commands with [Codecov.io Flags](https://docs.codecov.com/docs/flags). Effectively, this allows us to combine multiple reports which are run at various stages of our CI Pipeline or run as part of a parallel process.
+Line Code Coverage is generated by our unit tests and e2e tests, then combined by ([Codecov.io Flags](https://docs.codecov.com/docs/flags)), and finally reported in Github PRs by Codecov.io's PR Bot. This workflow gives a comprehensive (if flawed) view of line coverage.
 
-This e2e coverage is combined with our unit test report to give a comprehensive (if flawed) view of line coverage.
+### Karma-istanbul
+
+Line coverage is generated by our `karma-coverage-istanbul-reporter` package as defined in our `karma.conf.js` file:
+
+```js
+    coverageIstanbulReporter: {
+      fixWebpackSourcePaths: true,
+      skipFilesWithNoCoverage: true,
+      dir: 'coverage/unit', //Sets coverage file to be consumed by codecov.io
+      reports: ['lcovonly']
+    },
+```
+
+Once the file is generated, it can be published to codecov with
+
+```json
+    "cov:unit:publish": "codecov --disable=gcov -f ./coverage/unit/lcov.info -F unit",
+```
+
+### e2e
+
+The e2e line coverage is a bit more complex than the karma implementation. M 
+
+It generally follows the following flow:
+1. Each e2e suite will start the webpack.coverage.js config with the `npm run start:coverage` command which configures webpack with the `babel-plugin-istanbul` babel-loader to generate code coverage during e2e test execution.
```

Comment:
I know this is valid markdown, but let's number these 1, 2, 3, 4.. etc. Just so that if someone is not reading this in a markdown viewer they don't get confused.

Reply:
I will fight you over this... The markdown linter pr

Reply:
I'm not sure I'm up for this fight hahahah